Helper/Data.php

```
@@ -575,6 +578,44 @@ public function getApiKey()
         return $apiKey;
     }
 
+    /**
+     * Returns Environment URL from settings
+     *
+     * If we can not find a value
+     * @return string
+     */
+    public function getEnvironmentUrl(): string
+    {
+        $environmentUrl = $this->config->getValue(
+            'payment/divido_financing/environment_url',
+            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+        );
+
+        // If there is an url, use that.
+        if(!empty($environmentUrl)){
+            return $environmentUrl;
+        }
+
+        // Try to get environment URL from API key.
+        $apiKey = $this->getApiKey();
+        $env = $this->getEnvironment($apiKey);
+        if ($this->debug()) {
+            $this->logger->info('Getting Default Base URL for DividoFinancing, Env: ' . $env);
+        }
+
+        $environmentUrl = \Divido\MerchantSDK\Environment::CONFIGURATION[$env]['base_uri'];
+
+        if(!is_string($environmentUrl)){
+            if($this->debug()){
+                $this->logger->info('Could not determine URL for DividoFinancing');
+            }
+
+            return '';
```

should we throw an error instead of returning an empty string?